src/update-expression.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+/**
+ * Edit an existing item's attributes, or add a new item to a table if it does not already exist.
+ * You can put or add attribute values.
+ * @module Helpers
+ */
+
+const { ifElse, endsWith, compose, evolve, concat, dropLast, merge, flip } = require('ramda');
+
+/**
+ * @private
+ * because original concat sucks
+ */
+const catcon = flip(concat);
+
+/**
+ * Returns a function that, when invoked with a set of default values for a dynamo expression,
+ * it creates the mandatory UpdateExpression, AttributeNames and AttributeValues needed to build
+ * the `action` operations created by the given `operation` function.
+ *
+ * @function
+ * @param {String} action to execute in dynamodb, it can be SET, ADD, DELETE or REMOVE
+ * @param {Function} operations that returns an object with the operations to execute and their parameters
+ */
+const updateExpression = (action, operations) =>
```

Comment:
@crash7 If these functions are going to be _exported_ (i.e.: part of the API), mind turning them into actual named `function` with a `return` statement? Let's help debugging.